### DISCUSSION TITLE
Parts of page as responses

### DISCUSSION THREAD

**Discussion by AndiLeni** [unpoly/unpoly#484] [Q&A] [answered]
Hello,

I have a general question.
With HTMX it is enough to send html parts back to the browser.
Is this also possible with unpoly, instead of a whole html page?
For some cases, this seems more logical to me.

Thanks and kind regards

**Comment by adam12** [comment, marked as answer]
[It's possible](https://unpoly.com/optimizing-responses), but you would need to use the header that Unpoly sets ([`X-Up-Target`](https://unpoly.com/X-Up-Target)) as the hint on what to send back.

Henning did a presentation a long time ago (2016?) that has [a slide that covers this](http://triskweline.de/unpoly-rugb/#/42) as well.

**Reply by AndiLeni** [reply]
Thanks, I will have a look at these pages!

**Comment by moozeek** [comment]
In my experience it's fine to just send back the target's HTML.

So if your target is #somediv it's OK to just send the HTML of that element (no html, body etc.)
```html
<div id="somediv">
	<!-- your html -->
</div>
```
You only need to check the Unpoly headers if you want your backend to figure out whether to send the whole page or just the target.
